config/config-api/src/com/thoughtworks/go/config/JobConfig.java

```diff
@@ -451,6 +451,8 @@ private void setJobRunTypeAttribute(Map attributesMap) {
 	}
 
     public void validateNameUniqueness(Map<String, JobConfig> visitedConfigs) {
+        if (name() == null) return;
```

Comment:
could use `StringUtil.isBlank` instead. No need to go through the rest of it if name is an empty string too.